flexmeasures/ui/templates/base.html

```diff
@@ -712,6 +712,86 @@
         });
     </script>
 
+    <!-- JS to enable toast -->
+    <script defer>
+        function initiateToastCloseBtn(closeToastBtn){
+            console.log("CloseBtn is initiated")
+            // hide button
+            closeToastBtn.style.display = "none";
+
+            closeToastBtn.addEventListener("click", function () {
+                const toastElements = document.querySelectorAll(".toast");
+                toastElements.forEach((toast) => {
+                const toastInstance = bootstrap.Toast.getInstance(toast); // Get the toast instance
+                if (toastInstance) {
+                    // destroy the toast
+                    toastInstance.dispose();
+                    toast.remove();
+                }
+                });
+                // Hide the close button
+                closeToastBtn.style.display = "none";
+            });
+        }
+
+        function showToast(message, type, toastStack, closeToastBtn) {
+            console.log("showToast clicked")
+            let colorClass;
+            let colorStyle = "";
+            let title;
+        
+            // Determine the type of toast
+            if (type == "error") {
+              colorClass = "bg-danger";
+              title = "Error";
+            } else if (type == "success") {
+              colorClass = "bg-success";
+              title = "Success";
+            } else {
+              colorStyle =
+                "background-color: {{ primary_color }};";
+              title = "Info";
+            }
+        
+            // Create the toast HTML
+            const toast = document.createElement("div");
+            toast.classList.add("toast", "hide", "mb-1");
+            toast.setAttribute("data-bs-autohide", "true");
+            toast.setAttribute("role", "alert");
+            toast.setAttribute("aria-live", "assertive");
+            toast.setAttribute("aria-atomic", "true");
+        
+            toast.innerHTML = `
+              <div class="toast-header">
+                <div class="rounded me-2 ${colorClass}" style="width: 20px; height: 20px; display: inline-block; ${colorStyle}"></div>
+                <strong class="me-auto">${title}</strong>
+                <button type="button" class="btn-close" data-bs-dismiss="toast" aria-label="Close"></button>
+              </div>
+              <div class="toast-body">
+                ${message}
+              </div>
+            `;
+        
+            // Append toast to the toast stack
+            toastStack.insertAdjacentElement("afterbegin", toast);
+        
+            // Show the close all button
+            closeToastBtn.style.display = "block";
+        
+            // Initialize the toast
+            const toastInstance = new bootstrap.Toast(toast);
+            toastInstance.show();
+        
+            // Remove the toast from the DOM once it’s hidden
```

Comment:
Actually, I would not do that. I liked that toasts only are destroyed by an active user action (click close button).

Reply:
Could you clarify a bit more? I'm a bit confused. I am removing/destroying the toast as subsequent toasts will reveal all te other hidden toast which will not be a really good user experience as there will eventually be a very long stack of toasts

Reply:
I prefer that experience, as they are auto-hiding (which is convenient) and I might have missed what went wrong earlier. I can expressively close all of them easily now with that button you made.

Reply:
Oh, i see. So in that case, I should only destroy/remove them when the close btn is clicked?

Reply:
This modification has been made